js/apps/admin-ui/src/groups/components/MoveDialog.tsx

```diff
@@ -11,39 +11,13 @@ type MoveDialogProps = {
   refresh: () => void;
 };
 
-const moveToRoot = async (source: GroupRepresentation) => {
-  await adminClient.groups.del({ id: source.id! });
-  const { id } = await adminClient.groups.create({
-    ...source,
-    id: undefined,
-  });
-  if (source.subGroups) {
-    await Promise.all(
-      source.subGroups.map((s) =>
-        adminClient.groups.setOrCreateChild(
-          { id: id! },
-          {
-            ...s,
-            id: undefined,
-          }
-        )
-      )
-    );
-  }
-};
+const moveToRoot = (source: GroupRepresentation) =>
+  adminClient.groups.create(source);
 
 const moveToGroup = async (
   source: GroupRepresentation,
   dest: GroupRepresentation
-) => {
-  try {
-    await adminClient.groups.setOrCreateChild({ id: dest.id! }, source);
-  } catch (error: any) {
-    if (error.response) {
-      throw error;
-    }
-  }
-};
+) => adminClient.groups.setOrCreateChild({ id: dest.id! }, source);
```

Comment:
Removing the try catch here will cause the exception to be caught on line 31 below, which "handles" all exceptions by displaying an error in the UI. This is not desirable, since the operation actually succeeds. The problem originates in the file js/libs/keycloak-admin-client/src/resources/agent.ts, requestWithParams function. If the returnResourceIdInLocationHeader is set in that function, it expects the HTTP response to include a Location header, from which it can extract the generated resource Id. The adminClient.groups.create(...) and adminClient.groups.setOrCreateChild(...) methods, which are implemented in js/libs/keycloak-admin-client/src/resources/groups.ts, both set the returnResourceIdInLocationHeader parameter, because a new resource is created in some cases, and thus the Id needs to be retrieved. If you want to avoid an exception being thrown in the case where a new resource is not to be created, such as when an existing resource is passed in to these methods to be updated, then the code in the requestWithParams method should be modified to check for this condition. Unless that change is made, the original try-catch in this file should be kept to avoid displaying an error when there is none.